src/metabase/routes.clj

```diff
@@ -31,20 +33,31 @@
 (defn- load-template [path variables]
   (stencil/render-string (load-file-at-path path) variables))
 
+(defn- load-localization []
+  (log/info *locale*)
+  (if (and *locale* (not= (str *locale*) "en"))
+    (try
+      (load-file-at-path (str "frontend_client/app/locales/" *locale* ".json"))
+    (catch Throwable e
+      (log/warn (str "Locale " *locale* " not found."))
+      "null"))
+    "null"))
+
 (defn- entrypoint [entry embeddable? {:keys [uri]}]
   (-> (if (init-status/complete?)
         (load-template (str "frontend_client/" entry ".html")
-                       {:bootstrap_json (escape-script (json/generate-string (public-settings/public-settings)))
-                        :uri            (escape-script (json/generate-string uri))
-                        :base_href      (escape-script (json/generate-string (base-href)))
-                        :embed_code     (when embeddable? (embed/head uri))})
+                       {:bootstrap_json    (escape-script (json/generate-string (public-settings/public-settings)))
+                        :localization_json (escape-script (load-localization))
+                        :uri               (escape-script (json/generate-string uri))
+                        :base_href         (escape-script (json/generate-string (base-href)))
+                        :embed_code        (when embeddable? (embed/head uri))})
         (load-file-at-path "frontend_client/init.html"))
       resp/response
       (resp/content-type "text/html; charset=utf-8")))
 
-(def ^:private index  (partial entrypoint "index"  (not :embeddable)))
-(def ^:private public (partial entrypoint "public" :embeddable))
-(def ^:private embed  (partial entrypoint "embed"  :embeddable))
+(def ^:private index  (locale-negotiator (partial entrypoint "index"  (not :embeddable))))
```

Comment:
It seems like we'd want to apply the local negotiator to all routes, not just index/public/embed. Is there a reason to limit it?